ext/authx/authx.php

```diff
@@ -138,6 +138,14 @@ function authx_civicrm_config(&$config) {
  */
 function authx_civicrm_install() {
   _authx_civix_civicrm_install();
+
+  // If the system is already using HTTP `Authorization:` headers before installation, then
+  // it's probably an extra/independent layer of security.
+  // Only activate support for `Authorization:` if this looks like a clean/amenable environment.
+  // @link https://github.com/civicrm/civicrm-core/pull/22837
+  if (empty($_SERVER['HTTP_AUTHORIZATION']) && !Civi::settings()->getExplicit('authx_header_cred')) {
+    Civi::settings()->set('authx_header_cred', ['jwt', 'api_key']);
```

Comment:
> (@davejenx)  If `authx_header_cred` had been explicitly set to an empty array and then the extension is re-enabled from cli (

Ah, I see what you're saying. `getExplicit('authx_header_cred')` could return `NULL` or `[]` (ie "not explicitly set" or "empty leftover from previous installation*). Both are FALSE-ish.

I'm a little torn:

* The intent of this line is "if it's not explicitly set, then set it".  That would be better reflected by `NULL !== Civi::settings()->getExplict(...)`.
* There's a bigger issue around `disable`/`uninstall`/`setting`s. Comparing `disable`/`uninstall`, a major goal of `uninstall` is to wipe the slate clean -- uninstall should wipe the settings. But I think it's quite rare for anyone to implement `hook_uninstall` and cleanup old settings.

If there was a general fix for `uninstall`/`setting`s, then this conditional wouldn't matter.

I think maybe I erred in putting this under `hook_install`. Rather, `hook_enable` is probably a better fit. (To wit: This check should run for both new/clean-installs and existing/dirty re-activations. And it should be a `NULL` check.)

Reply:
I'll rebase/revise/`r-run` a couple of those edge-cases for the install-logic. After that, I'll add `merge ready` (since the main logic worked for @davejenx and @MattTrim1)